src/main/java/StringManipulation.java

```diff
@@ -1,33 +1,115 @@
+//Author: Jonathan Jarman
+//Date: 6/11/2023
+//File: String Manipulation
+/*The StringManipulation class provides a set of methods for manipulating strings,
+such as counting the number of words, removing characters at specific positions,
+extracting substrings, and restoring shuffled strings. It implements the StringManipulationInterface,
+which defines the contract for these operations.
+* */
 public class StringManipulation implements StringManipulationInterface {
 
+    String string; // holding string input
+
     @Override
-    public String getString() {
-        return null;
+    public String getString() { // method to return string
+        return string;
     }
 
     @Override
-    public void setString(String string) {
+    public void setString(String string) { // method to set the string
+        this.string = string;
     }
 
+    // this method will count the # of words in the string
     @Override
     public int count() {
-        return 0;
+        // throw null pointer exception if string is null
+        if (string == null) {
+            throw new NullPointerException("String is null");
+        }
+        // if string is empty throw an exception
+        if (string.isEmpty()) {
+            throw new IndexOutOfBoundsException("string is empty!");
+        }
+        // creating an array for the string
+        String[] words = string.split("\\s+");
+        // returning length of words(string count)
+        return words.length;
     }
 
     @Override
     public String removeNthCharacter(int n, boolean maintainSpacing) {
-        return null;
+        // if n < 0, throw an error
+        if (n <= 0) {
+            throw new IllegalArgumentException("'n' must be greater than zero.");
+        }
+        // if n > length, throw an error
+        if (n > string.length()) {
+            throw new IndexOutOfBoundsException("'n' is greater than the string length.");
+        }
+        // catching null errors
+        if (string == null) {
```

Comment:
if you want to check for null, it has to be before the string is accessed for its length (line 47), or the null exception will already be thrown. I would recommend just removing this line, as null access checking is already built in to strings

Reply:
edited order.